### PR TITLE
Prevent php notices when edit donation form

### PR DESCRIPTION
**Pull request by ravinderk**
Resolves https://github.com/impress-org/givewp/pull/5973#issuecomment-947971991

## Description

I find out that the admin can still see PHP notice about undefined settings in saved options if the Legacy form template selected. I updated the logic to prevent PHP notices.

## Testing Instructions

You should not see PHP notice:
- When creating a new donation form
- When editing donation form with Multi-step form template selected.
- When editing donation form with Legacy form template selected.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: src/Helpers/Form/Template.php [hunk 1]
```diff
@@ -82,10 +82,14 @@ public static function saveOptions( $formId, $settings ) {
 	 * @param array $settings
 	 */
 	public static function handleOptionsBackwardCompatibility( &$settings ) {
```
[the remark this answers is not in the document]
Comment (ravinderk) [reply src/Helpers/Form/Template.php]: fixed: https://github.com/impress-org/givewp/compare/829a0f710555...bcdda0b537a2

File: src/Helpers/Form/Template.php [hunk 2]
```diff
@@ -82,10 +82,14 @@ public static function saveOptions( $formId, $settings ) {
...
-		if ( ! isset( $settings['visual_appearance'] ) ) {
+		if (
+			! isset( $settings['visual_appearance'] ) &&
```
[the remark this answers is not in the document]
Comment (ravinderk) [reply src/Helpers/Form/Template.php]: fixed: https://github.com/impress-org/givewp/compare/bcdda0b537a2...2a7e48d6a8cd

**Comment by ravinderk** [comment]
@alaca  this pull request is ready to review.

**Review by alaca — APPROVED** [review]
Looks good, @ravinderk :+1:

**Pull request merged**